gauge not worked?

```
panic: inconsistent label cardinality

goroutine 1 [running]:
github.com/carlpett/stream_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*MetricVec).WithLabelValues(0xc420013080, 0xc4201ac2b0, 0x2, 0x2, 0x0, 0x0)
        /go/src/github.com/carlpett/stream_exporter/vendor/github.com/prometheus/client_golang/prometheus/vec.go:135 +0x8e
github.com/carlpett/stream_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*GaugeVec).WithLabelValues(0xc4201c81b0, 0xc4201ac2b0, 0x2, 0x2, 0x0, 0x0)
        /go/src/github.com/carlpett/stream_exporter/vendor/github.com/prometheus/client_golang/prometheus/gauge.go:108 +0x4c
github.com/carlpett/stream_exporter/linemetrics.GaugeVecLineMetric.MatchLine(0xc420017c80, 0x8, 0xc4201082a0, 0x21, 0xc420108510, 0x0, 0xc420017e48, 0x5, 0xc420017f60, 0x5, ...)
        /go/src/github.com/carlpett/stream_exporter/linemetrics/gauge.go:26 +0x173
github.com/carlpett/stream_exporter/linemetrics.(*GaugeVecLineMetric).MatchLine(0xc4200f02a0, 0xc4201b8150, 0x63)
        <autogenerated>:11 +0x9d
main.main()
        /go/src/github.com/carlpett/stream_exporter/exporter.go:130 +0xbd9
```

Hi @carlpett, Label "test-mongo" not unique?
run option:
` -input.type syslog -input.syslog.listenaddr 0.0.0.0:1555 -input.syslog.linetemplate '[tag] [content]' -input.syslog.listenfamily udp -config /app/stream_exporter.yaml -log.level debug`

```
metrics:
  - name: commited
    kind: gauge
    pattern: '(?P<section>test\-mongo).*total (?P<value>\d+)'
```
input stream syslog line:
```
DEBU[0022] Got line "test-mongo 2018-09-12 07:28:00,655 [DEBUG] testCopier : [9de2afd4] Committed for 1 to total 29395445600"  source="exporter.go:127"
```

```
  DEBU[0000] Got line "test-mongo 2018-09-12 07:28:18,961 [DEBUG] testCopier : [9de2afd4] Committed for 2 to total 29395445639"  source="exporter.go:127"
panic: inconsistent label cardinality

goroutine 1 [running]:
github.com/carlpett/stream_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*MetricVec).WithLabelValues(0xc42010e940, 0xc420182310, 0x2, 0x2, 0x0, 0x0)
        /go/src/github.com/carlpett/stream_exporter/vendor/github.com/prometheus/client_golang/prometheus/vec.go:135 +0x8e
github.com/carlpett/stream_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*GaugeVec).WithLabelValues(0xc42022a0d0, 0xc420182310, 0x2, 0x2, 0x0, 0x0)
        /go/src/github.com/carlpett/stream_exporter/vendor/github.com/prometheus/client_golang/prometheus/gauge.go:108 +0x4c
github.com/carlpett/stream_exporter/linemetrics.GaugeVecLineMetric.MatchLine(0xc4200fc580, 0x8, 0xc420110870, 0x2b, 0xc420110ae0, 0x0, 0xc4200fc860, 0x7, 0xc4200fc868, 0x7, ...)
        /go/src/github.com/carlpett/stream_exporter/linemetrics/gauge.go:26 +0x173
github.com/carlpett/stream_exporter/linemetrics.(*GaugeVecLineMetric).MatchLine(0xc4201362a0, 0xc420212100, 0x71)
        <autogenerated>:11 +0x9d
main.main()
        /go/src/github.com/carlpett/stream_exporter/exporter.go:130 +0xbd9
```

--- a/linemetrics/gauge.go
+++ b/linemetrics/gauge.go
@@ -16,14 +16,11 @@ type GaugeVecLineMetric struct {
 func (gauge GaugeVecLineMetric) MatchLine(s string) {
 	matches := gauge.pattern.FindStringSubmatch(s)
 	if len(matches) > 0 {
-		captures := matches[1:]
-		valueStr := captures[gauge.valueIdx]
-		value, err := strconv.ParseFloat(valueStr, 64)
+		labels, value, err := getLabelsAndValue(matches, gauge.valueIdx)
 		if err != nil {
-			log.Warnf("Unable to convert %s to float\n", valueStr)
 			return
 		}
-		gauge.metric.WithLabelValues(captures...).Set(value)
+		gauge.metric.WithLabelValues(labels...).Set(value)
 	}
 }
 

--- a/linemetrics/histogram.go
+++ b/linemetrics/histogram.go
@@ -36,15 +36,11 @@ type HistogramVecLineMetric struct {
 func (histogram HistogramVecLineMetric) MatchLine(s string) {
 	matches := histogram.pattern.FindStringSubmatch(s)
 	if len(matches) > 0 {
-		captures := matches[1:]
-		valueStr := captures[histogram.valueIdx]
-		value, err := strconv.ParseFloat(valueStr, 64)
+		labels, value, err := getLabelsAndValue(matches, histogram.valueIdx)
 		if err != nil {
-			log.Warnf("Unable to convert %s to float\n", valueStr)
 			return
 		}
-		capturedLabels := append(captures[0:histogram.valueIdx], captures[histogram.valueIdx+1:]...)
-		histogram.metric.WithLabelValues(capturedLabels...).Observe(value)
+		histogram.metric.WithLabelValues(labels...).Observe(value)
 	}
 }
 

--- a/linemetrics/linemetrics.go
+++ b/linemetrics/linemetrics.go
@@ -4,8 +4,10 @@ import (
 	"errors"
 	"io/ioutil"
 	"regexp"
+	"strconv"
 
 	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
 	"gopkg.in/yaml.v2"
 )
 
@@ -83,4 +85,16 @@ func getValueCaptureIndex(labels []string) (int, error) {
 	}
 
 	return valueIdx, nil
+}
+
+func getLabelsAndValue(matches []string, valueIdx int) ([]string, float64, error) {
+	captures := matches[1:]
+	valueStr := captures[valueIdx]
+	value, err := strconv.ParseFloat(valueStr, 64)
+	if err != nil {
+		log.Warnf("Unable to convert %s to float\n", valueStr)
+		return nil, 0, err
+	}
+	labels := append(captures[0:valueIdx], captures[valueIdx+1:]...)
+	return labels, value, nil
 }

--- a/linemetrics/summary.go
+++ b/linemetrics/summary.go
@@ -36,15 +36,11 @@ type SummaryVecLineMetric struct {
 func (summary SummaryVecLineMetric) MatchLine(s string) {
 	matches := summary.pattern.FindStringSubmatch(s)
 	if len(matches) > 0 {
-		captures := matches[1:]
-		valueStr := captures[summary.valueIdx]
-		value, err := strconv.ParseFloat(valueStr, 64)
+		labels, value, err := getLabelsAndValue(matches, summary.valueIdx)
 		if err != nil {
-			log.Warnf("Unable to convert %s to float\n", valueStr)
 			return
 		}
-		capturedLabels := append(captures[0:summary.valueIdx], captures[summary.valueIdx+1:]...)
-		summary.metric.WithLabelValues(capturedLabels...).Observe(value)
+		summary.metric.WithLabelValues(labels...).Observe(value)
 	}
 }
 

